azurerm/internal/services/web/app_service.go

```diff
@@ -91,6 +91,32 @@ func schemaAppServiceGoogleAuthSettings() *schema.Schema {
 	}
 }
 
+func schemaAppServiceGithubAuthSettings() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"client_id": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
```

Comment:
Should we be validating this?

Reply:
pulled this in favour of adding as a separate PR.